src/lib.rs

```diff
@@ -1,88 +1,298 @@
+    missing_docs,
+    trivial_casts,
+    trivial_numeric_casts,
+    unused_import_braces,
+    unused_qualifications
 )]
 #![cfg(target_os = "macos")]
 #![allow(improper_ctypes)]
 
-extern crate chrono;
-extern crate objc_foundation;
 pub mod error;
 
 use chrono::offset::*;
-use error::{ ApplicationError, NotificationError, NotificationResult };
-use objc_foundation::{INSString, NSString};
+use error::{ApplicationError, NotificationError, NotificationResult};
+use objc_foundation::{INSDictionary, INSString, NSDictionary, NSString};
+use objc_id::Id;
 use std::ops::Deref;
 use std::path::PathBuf;
 
 static mut APPLICATION_SET: bool = false;
 
 mod sys {
-    use objc_foundation::NSString;
+    use objc_foundation::{NSDictionary, NSString};
+    use objc_id::Id;
     #[link(name = "notify")]
     extern "C" {
-        pub fn scheduleNotification(
-            title: *const NSString,
-            subtitle: *const NSString,
-            message: *const NSString,
-            sound: *const NSString,
-            deliveryDate: f64,
-        ) -> bool;
         pub fn sendNotification(
             title: *const NSString,
             subtitle: *const NSString,
             message: *const NSString,
-            sound: *const NSString,
-        ) -> bool;
+            options: *const NSDictionary<NSString, NSString>,
+        ) -> Id<NSDictionary<NSString, NSString>>;
         pub fn setApplication(newbundleIdentifier: *const NSString) -> bool;
         pub fn getBundleIdentifier(appName: *const NSString) -> *const NSString;
     }
 }
 
-/// Schedules a new notification in the NotificationCenter
-///
-/// Returns a `NotificationError` if a notification could not be scheduled
-/// or is scheduled in the past
-///
-/// # Example:
-///
-/// ```ignore
-/// extern crate chrono;
-/// # use mac_notification_sys::*;
-/// use chrono::offset::*;
-///
-/// // schedule a notification in 5 seconds
-/// let _ = schedule_notification("Title", &None, "This is the body", &Some("Ping"),
-///                               Utc::now().timestamp() as f64 + 5.).unwrap();
-/// ```
-pub fn schedule_notification(
-    title: &str,
-    subtitle: &Option<&str>,
-    message: &str,
-    sound: &Option<&str>,
-    delivery_date: f64,
-) -> NotificationResult<()> {
-    ensure!(
-        delivery_date >= Utc::now().timestamp() as f64,
-        NotificationError::ScheduleInThePast
-    );
-
-    let use_sound = match sound {
-        Some(sound) if check_sound(sound) => sound,
-        _ => "_mute",
-    };
-    unsafe {
-        ensure!(
-            sys::scheduleNotification(
-                NSString::from_str(title).deref(),
-                NSString::from_str(subtitle.unwrap_or("")).deref(),
-                NSString::from_str(message).deref(),
-                NSString::from_str(use_sound).deref(),
-                delivery_date,
+/// Possible actions accessible through the main button of the notification
+pub enum MainButton<'a> {
```

Comment:
Maybe we should move all the new enums and structs and their implementations into a new file. So the lib.rs is clean again.